base/cvd/cuttlefish/host/commands/cvd/common_utils.cpp

```diff
@@ -30,6 +30,32 @@
 
 namespace cuttlefish {
 
+namespace {
+/*
+ * Most branches read the kAndroidHostOut environment variable, but a few read
+ * kAndroidSoongHostOut instead. Cvd will set both variables for the subtools
+ * to the first value it finds:
+ * - envs[kAndroidHostOut] if variable is set
+ * - envs[kAndroidSoongHostOut] if variable is set
+ * - envs["HOME"] if envs["HOME"] + "/bin/cvd_internal_start" exists.
+ * - current working directory
+ */
+std::string GetNewAndroidHostOut(const cvd_common::Envs& envs) {
+  if (Contains(envs, kAndroidHostOut)) {
+    return envs.at(kAndroidHostOut);
+  }
+  if (Contains(envs, kAndroidSoongHostOut)) {
+    return envs.at(kAndroidSoongHostOut);
+  }
+  if (Contains(envs, "HOME") &&
+      FileExists(envs.at("HOME") + "/bin/cvd_internal_start")) {
+    return envs.at("HOME");
+  }
+  std::unique_ptr<char, void (*)(void*)> cwd(getcwd(nullptr, 0), &free);
```

Comment:
`files.h::CurrentDirectory()` already implements this the same way (with a little logging if there is a failure)

Reply:
Done